sklearn/linear_model/ridge.py

```diff
@@ -870,10 +879,16 @@ def _values(self, alpha, y, v, Q, QT_y):
         G_diag, c = self._errors_and_values_helper(alpha, y, v, Q, QT_y)
         return y - (c / G_diag), c
 
-    def _pre_compute_svd(self, X, y):
+    def _pre_compute_svd(self, X, y, centered_kernel=True):
         if sparse.issparse(X):
             raise TypeError("SVD not supported for sparse matrices")
-        U, s, _ = linalg.svd(X, full_matrices=0)
+        if centered_kernel:
+            X_ = np.hstack((X, np.ones((X.shape[0], 1))))
+        else:
+            X_ = X
```

Comment:
You don't need the intermediate variable "X_". You can just do "X = np.hstack(...", it won't modify the original X that will keep living outside the function.